tests/test_graph_components/providers/test_domain_provider.py

```diff
@@ -0,0 +1,37 @@
+from pathlib import Path
+from typing import Text
+
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.graph_components.providers.domain_provider import DomainProvider
+from rasa.shared.core.domain import Domain
+from rasa.shared.importers.importer import TrainingDataImporter
+
+
+def test_domain_provider_generates_and_persists_domain(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    tmp_path: Path,
+    config_path: Text,
+    domain_path: Text,
+):
+    resource = Resource.from_cache("xy", tmp_path, default_model_storage)
```

Comment:
You can just use the constructor. The `from_cache` parts is basically always off limits as this is only called by the `TrainingHook`
```suggestion
    resource = Resource("some resource")
```